config/v1/types_infrastructure.go

```diff
@@ -144,6 +148,19 @@ type GCPPlatformStatus struct {
 	Region string `json:"region"`
 }
 
+// BareMetalPlatformStatus holds the current status of the BareMetal infrastructure provider.
+type BareMetalPlatformStatus struct {
+	// apiVIP holds the Virtual IP (VIP) that can be used to contact the external OpenShift API
+	APIVIP string `json:"apiVIP"`
+
+	// ingressVIP holds the Virtual IP (VIP) for the Ingress operator
+	IngressVIP string `json:"ingressVIP"`
+
+	// dnsVIP holds the Virtual IP (VIP) for the CoreDNS running in the cluster. This
+	// CoreDNS will be serving hostname for the BareMetal hosts in the cluster.
+	DNSVIP string `json:"dnsVIP"`
```

Comment:
@ironcladlou you aware of this and is is the doc correct?

Reply:
The dns-operator publishes the DNS service clusterIP though the `operator.openshift.io/dns` [status field](https://github.com/openshift/api/blob/master/operator/v1/types_dns.go#L52).

What would set/manage this new field? Who consumes it?

Reply:
Does this field refer to some CoreDNS instance _other than_ the one managed by dns-operator?

Reply:
yes, these CoreDNS servers are in static pods managed set by MCO

Reply:
>yes, these CoreDNS servers are in static pods managed set by MCO

In that case, I suggest the comment needs some elaboration, like:

> dnsVIP is the IP of a special-purpose cluster DNS service used only for resolving node host names in $some_contexts.

Or something else to disambiguate this from the general cluster DNS service used for pod DNS resolution/service discovery, which is totally separate.

Other small notes:

* CoreDNS is an implementation detail — isn't the service just a DNS server?
* "Virtual" IPs seem like an implementation detail

Reply:
Yes, it is a DNS server. I can remove the CoreDNS references.

I think I'm starting to see your point about the VIP being an implementation detail.

I thought with my latest patch I had disambiguated its purpose, but I'll take another stab at it.

Reply:
IP vs VIP - either one is fine, to me.

This is indeed a configuration item for bare metal IPI clusters.  It's an IP that is reserved for our use for this internal DNS service.

Reply:
Is "vip" an implementation detail though?  As far as I have gathered from the discussion so far and the referenced documentation at https://github.com/openshift-metal3/kni-installer/blob/master/docs/design/baremetal/networking-infrastructure.md, the purpose of this API is to communicate to MCO, "I need keepalived configured to use this vip for the API, that vip for DNS, and that other vip for the default ingress controller"—nothing else but MCO is using this API, and this API is not being used for any purpose but to specify configuration for keepalived, right?

Reply:
I keep thinking the 3 fields should be replaced by a slice of vips.  Any reason MCO needs to be told which vip to use for which component?

Reply:
I would say so. These VIPs will also have external datacenter DNS records. So, it would be important of which VIP is which.

Reply:
My only point was to make sure the word "virtual" was actually significant. Given just an IP address, how can you know it's virtual, why would you care as a consumer, and why would the API need to make a statement about the IPs use in a virtual IP mechanism?

So I was just bike shedding on the cosmetics. I wondered whether these are simply "IP" addresses rather than "virtual IP" addresses.

Reply:
> Given just an IP address, how can you know it's virtual, why would you care as a consumer, and why would the API need to make a statement about the IPs use in a virtual IP mechanism?

Because it is the configuration parameter for the vip that MCO will configure.  The installer uses this API to tell MCO, "Configure a vip for DNS with this address."  It seems misleading to try to make the name generic in a way in which the API it is not.

Reply:
> I would say so. These VIPs will also have external datacenter DNS records. So, it would be important of which VIP is which.

Is it a requirement to determine the address ahead of installation (i.e., determine which vip is for DNS, which is for ingress, and which is for the API)?  Then yeah, it makes sense to use separate fields.  (To enable lookup of the address after installation, I would think a headless service would be most suitable.)

Reply:
They are used in the install process. They are being added to the installconfig survey.